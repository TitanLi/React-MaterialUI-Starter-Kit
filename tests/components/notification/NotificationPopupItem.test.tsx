/*eslint no-console: ["off", { allow: ["warn", "error"] }] */

import React from "react";
import { mount } from "enzyme";

import NotificationPopupItem from "../../../src/components/Notification/NotificationPopupItem";
import Type from "../../../src/components/Notification/NotificationType";

const render = ({ onClose, ...rest } = {}) => {
  if (!onClose) onClose = () => true;

  const item = mount(
    <NotificationPopupItem
      {...rest}
      onClose={onClose}
      message="Test Notification Item"
    />
  );

  if (item.mount) item.mount();
  return item;
};

const promiseComponent = ({ displayIn, ...rest }) =>
  new Promise((resolve, reject) => {
    let called = false;

    const item = render({
      ...rest,
      displayIn: displayIn,
      onClose: () => {
        called = true;
        resolve(called);
      }
    });

    setTimeout(() => {
      if (!called) reject(called);
    }, displayIn + 2);
  });

describe("Testing NotificationPopupItem component", () => {
  test("onClose will be called", () => {
    const pm = promiseComponent({ displayIn: 2 });
    return expect(pm).resolves.toBe(true);
  });

  test("if displayIn is 0 the onClose wont be called", () => {
    const pm = promiseComponent({ displayIn: 0 });
    return expect(pm).rejects.toBe(false);
  });

  test("update displayIn from 0 to 5 the onClose will be called", async () => {
    const item = render({ displayIn: 0 });

    //DisplayIn = 0
    const p1 = new Promise((resolve, reject) => {
      item.setProps({ onClose: () => resolve(true) });
      setTimeout(() => reject(false), 2);
    });
    //onClose called
    await expect(p1).rejects.toBe(false);

    //DisplayIn = 5ms
    const p2 = new Promise((resolve, reject) => {
      item.setProps({ displayIn: 2, onClose: () => resolve(true) });
    });

    //onClose called
    await expect(p2).resolves.toBe(true);
  });

  test("throw exception when set unknown Type", () => {
    const original = console.error;
    console.error = jest.fn();
    render({ type: "Hello" });

    expect(console.error).toHaveBeenCalled();
    console.error = original;
  });

  test("throw exception when set icon Type", () => {
    const original = console.error;
    console.error = jest.fn();
    render({ icon: "Hello" });

    expect(console.error).toHaveBeenCalled();
    console.error = original;
  });

  test("onClose must be a func", () => {
    const original = console.error;
    console.error = jest.fn();
    render({ onClose: "Hello" });

    expect(console.error).toHaveBeenCalled();
    console.error = original;
  });

  test("onClick must be a func", () => {
    const original = console.error;
    console.error = jest.fn();
    render({ onClick: "Hello" });

    expect(console.error).toHaveBeenCalled();
    console.error = original;
  });

  test("onClick test", () => {
    var onClick = jest.fn();
    const wrapper = render({ onClick });

    const btn = wrapper.findWhere(
      b => b.type() === "div" && b.prop("onClick") === onClick
    );

    expect(wrapper).toMatchSnapshot();
    //Message button and Close button
    expect(btn.length).toBe(1);

    btn.simulate("click");
    expect(onClick).toBeCalled();
  });

  test(`render ${Type.INFO}`, () => {
    const item = render({ type: Type.INFO });
    expect(item).toMatchSnapshot();
  });

  test(`render ${Type.CONFIRM}`, () => {
    const item = render({ type: Type.CONFIRM });
    expect(item).toMatchSnapshot();
  });

  test(`render ${Type.DANGER}`, () => {
    const item = render({ type: Type.DANGER });
    expect(item).toMatchSnapshot();
  });

  test(`render ${Type.SUCCESS}`, () => {
    const item = render({ type: Type.SUCCESS });
    expect(item).toMatchSnapshot();
  });

  test(`render ${Type.WARNING}`, () => {
    const item = render({ type: Type.WARNING });
    expect(item).toMatchSnapshot();
  });

  test(`render ${Type.WARNING} with custom Icon`, () => {
    const icon = <div id="icon">Duy</div>;
    const item = render({ type: Type.WARNING, icon });

    expect(item.find("#icon").length).toBe(1);
    expect(item).toMatchSnapshot();
  });

  test(`render ${Type.WARNING} with Icon is undefined`, () => {
    const item = render({ type: Type.WARNING, icon: undefined });
    expect(item).toMatchSnapshot();
  });
});
