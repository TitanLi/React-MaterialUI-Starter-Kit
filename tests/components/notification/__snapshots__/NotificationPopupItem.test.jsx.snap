// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NotificationPopupItem component onClick test 1`] = `
Array [
  <div
    className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-info-6"
      role="alertdialog"
      style={Object {}}
    >
      <div
        className="MuiSnackbarContent-message-31 Snackbar-message-12"
      >
        <div
          className=""
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-96 Snackbar-icon-15"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
            />
          </svg>
          <span
            className="Snackbar-iconMessage-22"
          >
            Test Notification Item
          </span>
        </div>
      </div>
      <div
        className="MuiSnackbarContent-action-32"
      >
        <button
          aria-label="Close"
          className="MuiButtonBase-root-111 MuiIconButton-root-105 MuiIconButton-colorInherit-106 Snackbar-iconButton-14"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-110"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-96 Snackbar-close-13"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-114"
          />
        </button>
      </div>
    </div>
  </div>,
  "",
]
`;

exports[`Testing NotificationPopupItem component render confirm 1`] = `
Array [
  <div
    className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-primary-10"
      role="alertdialog"
      style={Object {}}
    >
      <div
        className="MuiSnackbarContent-message-31 Snackbar-message-12"
      >
        <div
          className=""
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-96 Snackbar-icon-15"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
            />
          </svg>
          <span
            className="Snackbar-iconMessage-22"
          >
            Test Notification Item
          </span>
        </div>
      </div>
      <div
        className="MuiSnackbarContent-action-32"
      >
        <button
          aria-label="Close"
          className="MuiButtonBase-root-111 MuiIconButton-root-105 MuiIconButton-colorInherit-106 Snackbar-iconButton-14"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-110"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-96 Snackbar-close-13"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-114"
          />
        </button>
      </div>
    </div>
  </div>,
  "",
]
`;

exports[`Testing NotificationPopupItem component render danger 1`] = `
Array [
  <div
    className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-danger-9"
      role="alertdialog"
      style={Object {}}
    >
      <div
        className="MuiSnackbarContent-message-31 Snackbar-message-12"
      >
        <div
          className=""
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-96 Snackbar-icon-15"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
            />
          </svg>
          <span
            className="Snackbar-iconMessage-22"
          >
            Test Notification Item
          </span>
        </div>
      </div>
      <div
        className="MuiSnackbarContent-action-32"
      >
        <button
          aria-label="Close"
          className="MuiButtonBase-root-111 MuiIconButton-root-105 MuiIconButton-colorInherit-106 Snackbar-iconButton-14"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-110"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-96 Snackbar-close-13"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-114"
          />
        </button>
      </div>
    </div>
  </div>,
  "",
]
`;

exports[`Testing NotificationPopupItem component render info 1`] = `
Array [
  <div
    className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-info-6"
      role="alertdialog"
      style={Object {}}
    >
      <div
        className="MuiSnackbarContent-message-31 Snackbar-message-12"
      >
        <div
          className=""
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-96 Snackbar-icon-15"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
            />
          </svg>
          <span
            className="Snackbar-iconMessage-22"
          >
            Test Notification Item
          </span>
        </div>
      </div>
      <div
        className="MuiSnackbarContent-action-32"
      >
        <button
          aria-label="Close"
          className="MuiButtonBase-root-111 MuiIconButton-root-105 MuiIconButton-colorInherit-106 Snackbar-iconButton-14"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-110"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-96 Snackbar-close-13"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-114"
          />
        </button>
      </div>
    </div>
  </div>,
  "",
]
`;

exports[`Testing NotificationPopupItem component render success 1`] = `
Array [
  <div
    className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-success-7"
      role="alertdialog"
      style={Object {}}
    >
      <div
        className="MuiSnackbarContent-message-31 Snackbar-message-12"
      >
        <div
          className=""
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-96 Snackbar-icon-15"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          <span
            className="Snackbar-iconMessage-22"
          >
            Test Notification Item
          </span>
        </div>
      </div>
      <div
        className="MuiSnackbarContent-action-32"
      >
        <button
          aria-label="Close"
          className="MuiButtonBase-root-111 MuiIconButton-root-105 MuiIconButton-colorInherit-106 Snackbar-iconButton-14"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-110"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-96 Snackbar-close-13"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-114"
          />
        </button>
      </div>
    </div>
  </div>,
  "",
]
`;

exports[`Testing NotificationPopupItem component render warning 1`] = `
Array [
  <div
    className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-warning-8"
      role="alertdialog"
      style={Object {}}
    >
      <div
        className="MuiSnackbarContent-message-31 Snackbar-message-12"
      >
        <div
          className=""
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-96 Snackbar-icon-15"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
            />
          </svg>
          <span
            className="Snackbar-iconMessage-22"
          >
            Test Notification Item
          </span>
        </div>
      </div>
      <div
        className="MuiSnackbarContent-action-32"
      >
        <button
          aria-label="Close"
          className="MuiButtonBase-root-111 MuiIconButton-root-105 MuiIconButton-colorInherit-106 Snackbar-iconButton-14"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-110"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-96 Snackbar-close-13"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-114"
          />
        </button>
      </div>
    </div>
  </div>,
  "",
]
`;

exports[`Testing NotificationPopupItem component render warning with Icon is undefined 1`] = `
Array [
  <div
    className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-warning-8"
      role="alertdialog"
      style={Object {}}
    >
      <div
        className="MuiSnackbarContent-message-31 Snackbar-message-12"
      >
        <div
          className=""
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-96 Snackbar-icon-15"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
            />
          </svg>
          <span
            className="Snackbar-iconMessage-22"
          >
            Test Notification Item
          </span>
        </div>
      </div>
      <div
        className="MuiSnackbarContent-action-32"
      >
        <button
          aria-label="Close"
          className="MuiButtonBase-root-111 MuiIconButton-root-105 MuiIconButton-colorInherit-106 Snackbar-iconButton-14"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-110"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-96 Snackbar-close-13"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-114"
          />
        </button>
      </div>
    </div>
  </div>,
  "",
]
`;

exports[`Testing NotificationPopupItem component render warning with custom Icon 1`] = `
Array [
  <div
    className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-warning-8"
      role="alertdialog"
      style={Object {}}
    >
      <div
        className="MuiSnackbarContent-message-31 Snackbar-message-12"
      >
        <div
          className=""
        >
          <div
            className="Snackbar-icon-15"
          >
            <div
              id="icon"
            >
              Duy
            </div>
          </div>
          <span
            className="Snackbar-iconMessage-22"
          >
            Test Notification Item
          </span>
        </div>
      </div>
      <div
        className="MuiSnackbarContent-action-32"
      >
        <button
          aria-label="Close"
          className="MuiButtonBase-root-111 MuiIconButton-root-105 MuiIconButton-colorInherit-106 Snackbar-iconButton-14"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-110"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-96 Snackbar-close-13"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-114"
          />
        </button>
      </div>
    </div>
  </div>,
  "",
]
`;
